### PR TITLE
Hide log box in TopBar components

### PR DESCRIPTION
**Pull request by guyca**
Up until RN 62, YellowBox was used to show warnings. RNN was able to detect those and show them only on components used as screens.
RN 63 changed the default implementation to LogBox. With the switch to LogBox we were no longer able to hide warnings on TopBar components.

This commit hides LogBox in topBar components.

### CODE REVIEW

File: lib/android/app/src/reactNative55/java/com/reactnativenavigation/viewcontrollers/viewcontroller/YellowBoxHelper.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import static com.reactnativenavigation.utils.ViewUtils.findChildrenByClassRecursive;
+
+public class YellowBoxHelper {
+    private final static int YELLOW_BOX_COLOR = -218449360;
+
+    boolean isYellowBox(View parent, View child) {
+        return parent instanceof ViewGroup &&
+               child instanceof ViewGroup &&
+               ((ViewGroup) parent).getChildCount() > 1 &&
+               !findChildrenByClassRecursive((ViewGroup) child, View.class, YellowBackgroundMather()).isEmpty();
```
Comment (swabbass) [inline lib/android/app/src/reactNative55/java/com/reactnativenavigation/viewcontrollers/viewcontroller/YellowBoxHelper.java:20]: typo?

**Comment by swabbass** [comment]
@guyca  THX for the PR: looks fine.

Do we have a plan to support these old versions of RN for a long time? 51, 55, 56, 57, 57_5, 60, 62, 63, like we might schedule a deadline for the old RN versions we support, and keep the new ones?

File: lib/android/app/src/reactNative55/java/com/reactnativenavigation/viewcontrollers/viewcontroller/YellowBoxHelper.java (hunk 1 above)
Comment (guyca) [reply lib/android/app/src/reactNative55/java/com/reactnativenavigation/viewcontrollers/viewcontroller/YellowBoxHelper.java:20]: Fixed

**Comment by guyca** [comment]
> Do we have a plan to support these old versions of RN for a long time? 51, 55, 56, 57, 57_5, 60, 62, 63, like we might schedule a deadline for the old RN versions we support, and keep the new ones?

I think the next major should drop support for pre RN 63 version. These old version are holding us back.

**Review by yogevbd — APPROVED** [review]

**Pull request merged**